[HDInsight] Add kafka rest proxy feature

aim-for-better opened the pull request:
The kafka rest proxy feature is related to Spec PR [[HDInsight] Support kafka rest proxy feature ](https://github.com/Azure/azure-rest-api-specs/pull/7844)

1. Add new feature kafka rest proxy to HDInsight SDK
2. Add related session test and session records

Commit:
Add kafkaRestProxy feature and realted test

Commit:
Add session records for create kafka cluster with rest proxy

isra-fel reviewed (changes requested):
Hi @aim-for-better, could you
- Include the `.txt` file generated when you run `generate.ps1`[(details)](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#sdk-generation-from-updated-spec)
- Update api tag in `AzSdk.RP.props`, release notes in csproj
- Update versions in csproj and `assemblyinfo.cs` if you are going to release

Thanks

Commit:
1. Add .txt file generated when runing generate.ps1
2. update the release notes in csproj
3. update the version

aim-for-better commented:
> Hi @aim-for-better, could you
> 
> * Include the `.txt` file generated when you run `generate.ps1`[(details)](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#sdk-generation-from-updated-spec)
> * Update api tag in `AzSdk.RP.props`, release notes in csproj
> * Update versions in csproj and `assemblyinfo.cs` if you are going to release
> 
> Thanks

Hi @isra-fel,
Thanks for your comments.
- Added .txt
- Updated the release notes and version

I think that we don't need to modify AzSdk.Rp.props. I haven't changed the file in the previous PR.

isra-fel commented:
Yeah you're right. It needs to be changed only when the API version changes
